### PR TITLE
Issue 3533: confirmation form that all theme custom settings will be lost if disabled

### PR DESCRIPTION
**Pull request by herbdool**
Addresses https://github.com/backdrop/backdrop-issues/issues/3533

### CODE REVIEW

File: core/modules/system/system.admin.inc [hunk 1]
```diff
@@ -285,31 +285,46 @@ function system_theme_enable() {
...
+function system_theme_disable_confirm_submit($form, &$form_state) {
+  $theme = $form_state['complete_form']['theme'];
+  // Get current list of themes.
+  $themes = list_themes();
+
+  // Check if the specified theme is one recognized by the system.
+  if (!empty($themes[$theme])) {
+    if ($theme == config_get('system.core', 'theme_default')) {
+      // Don't disable the default theme.
+      backdrop_set_message(t('%theme is the default theme and cannot be disabled.', array('%theme' => $themes[$theme]->info['name'])), 'error');
```
Comment (klonos) [inline core/modules/system/system.admin.inc:316]: We are not rendering any "Disable" link in the theme listing page for the default theme, so this entire check and message seems to not be required.

File: core/modules/system/system.admin.inc [hunk 2]
```diff
@@ -285,31 +285,46 @@ function system_theme_enable() {
...
-    if (!empty($themes[$theme])) {
-      if ($theme == config_get('system.core', 'theme_default')) {
-        // Don't disable the default theme.
-        backdrop_set_message(t('%theme is the default theme and cannot be disabled.', array('%theme' => $themes[$theme]->info['name'])), 'error');
-      }
-      else {
-        theme_disable(array($theme));
-        backdrop_set_message(t('The %theme theme has been disabled.', array('%theme' => $themes[$theme]->info['name'])));
-      }
+    $confirm_question = t('Are you sure you want to disable this theme? All custom configuration will be lost.');
```
Comment (klonos) [inline core/modules/system/system.admin.inc:298]: The "All custom configuration will be lost." bit should not be part of the page title/question. It should be a warning message instead:

```suggestion
    $confirm_question = t('Are you sure you want to disable this theme?');
    backdrop_set_message(t('All custom configuration will be lost!'), 'warning');
    backdrop_set_message(t('This action cannot be undone.'), 'warning');
```

File: core/modules/system/system.admin.inc [hunk 3]
```diff
@@ -285,31 +285,46 @@ function system_theme_enable() {
...
+    return confirm_form($form, $confirm_question, 'admin/appearance', t('This action cannot be undone.'), t('Disable'), t('Cancel'));
```
Comment (klonos) [inline core/modules/system/system.admin.inc:299]:
```suggestion
    return confirm_form($form, $confirm_question, 'admin/appearance', NULL, t('Disable'), t('Cancel'));
```

File: core/modules/system/system.admin.inc (hunk 2 above)
Comment (herbdool) [reply core/modules/system/system.admin.inc:298]: This PR uses the same format as content type deletion confirmation. I would suggest keeping it consistent with that one. And change them in all confirmation pages in a separate PR.

File: core/modules/system/system.admin.inc (hunk 1 above)
Comment (herbdool) [reply core/modules/system/system.admin.inc:316]: I think the check and warning should stay. People could still make a url to disable it. This is no different from current situation where there isn't a disable link.

Comment (klonos) [reply core/modules/system/system.admin.inc:316]: Fair enough.

**Pull request merged**